lib/SIL/Utils/BasicBlockUtils.cpp

```diff
@@ -521,3 +521,36 @@ void swift::findJointPostDominatingSet(
     }
   }
 }
+
+//===----------------------------------------------------------------------===//
+//                          checkReachingBlockDominance
+//===----------------------------------------------------------------------===//
+
+#ifndef NDEBUG
+/// Verify that \p sourceBlock dominates \p destBlock given that some path from
+/// \p sourceBlock to \p destBlock exists.
+///
+/// Useful for assertions without computing Dominance.
+bool swift::checkReachingBlockDominates(SILBasicBlock *sourceBlock,
```

Comment:
Therefore, I suggest to call this utility `dominates` and add a comment that the complexity is O(numberOfBlocksInFunction).